docs/signature.md

```diff
@@ -0,0 +1,93 @@
+# Image Signature Specification
+
+**Version 0.1**
+
+## Introduction
+
+This document defines a detached container image signature object and signing methods.
+
+## Signature Format
+
+```js
+{
+    "critical": {/* required fields */
```

Comment:
`critical: []` is extra parsing; why would we want extra parsing in the most paranoid code?

Embedding the `critical`/`optional` distinction into the object structure also makes it an explicit design decision at specification creation time, and eliminates an entire class of mistakes (mistakenly marking a critical field as optional, or vice versa).

Honestly I’d much prefer most people to use the `github.com/containers/image/signature` library instead of parsing this themselves; the head scratching to understand `critical`/`optional` is nothing compared to the paranoia needed for https://github.com/containers/image/blob/master/signature/json.go and the like.

I’m not saying that `github.com/containers/image/signature` is perfect (definitely not, and adversarial reviews and help would be very welcome), but I do want consumers of signatures to resist the temptation to just call `json.Unmarshal` (or even worse, do something with `sed` in shell scripts), when most of us can share a single hardened implementation, including possibly some horrible workarounds depending on `optional.creator`.